crates/apollo-mcp-registry/src/platform_api/operation_collections/collection_poller.rs

```diff
@@ -77,26 +76,34 @@ async fn handle_poll_result(
         })
         .collect();
 
-    if changed_ids.is_empty() {
+    if changed_ids.is_empty() && removed_ids.is_empty() {
         tracing::debug!("no operation changed");
         Ok(None)
```

Comment:
Can we return early here, like with `return Ok(None);` to flatten the nested if statements below? Or is this not a good practice in Rust?

Reply:
I think thats fine! I'mm update